central/scannerdefinitions/handler/handler.go

```diff
@@ -140,6 +139,14 @@ func init() {
 	var err error
 	scannerUpdateBaseURL, err = url.Parse("https://definitions.stackrox.io")
 	utils.CrashOnError(err) // This is very unexpected.
+
+	// We try to detect and take into account only the version
+	// number (ie. X.Y.Z), otherwise copy the full version.
+	mainVersion := version.GetMainVersion()
+	mainVersionNum = mainVersionPattern.FindString(mainVersion)
+	if mainVersionNum == "" {
```

Comment:
33332e5a22

Reply:
OK, so I decided to match against all version variants similar to how `roxctl scanner download-db` does. It allows us to match versions such as `4.6.x` and also specific builds, if we would like. For nightly builds we would have to create a `scanner-vulns-4.6.x.zip` with definitions accepting releases such as `4.6.x`. This is not part of the scope right, so I am capturing it as a follow-up: https://issues.redhat.com/browse/ROX-26580

Reply:
Perhaps to consider in the future: store multiple offline bundles. This way, I don't think we have to do all the release version checks. We can just say upload works but then download fails because we don't have `v1` or `v2`. Does that make sense?

Reply:
I've originally considered that. Uploading multiple bundles might give users a false sense of success, as the failure will happen later when Scanner is pulling the data. There is another concern from David about how we could even try to validate the bundle pre-upload, in the client side. I think this is separate story, which is valid, and can bring better experience -- but for now I think expecting one valid definition from within the bundle reduces confusion and simplifies.